### ISSUE TITLE
[Thanos Chart] store.metrics.annotations.enabled failes

### ISSUE THREAD

**Issue by ebini** [banzaicloud/banzai-charts#1036]
setting:
```
store:
  metrics:
    annotations:
      enabled: true
```

will throw an error.

on other components no error is thrown.

Error: UPGRADE FAILED: error validating "": error validating data: [ValidationError(Deployment.spec.template.metadata): unknown field "prometheus.io/port" in io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta, ValidationError(Deployment.spec.template.metadata): unknown field "prometheus.io/scrape" in io.k8s.apimachinery.pkg.apis.meta.v1.ObjectMeta]

**Comment by michaelcrunch** [comment]
After my fix was merged for the deployment templates this is still not working for the store component, I can confirm that query and compact are now working as expected. 

I noticed that the store-deployment template uses a $root prefix which is not present in the other templates, I've attempted to remove it locally but cannot get the annotations working. 

I'm not sure if a maintainer could take a look at the template and see if I am missing something?
